packages/arb-token-bridge-ui/synpress.config.ts

```diff
@@ -98,14 +112,71 @@ export default defineConfig({
 
       // Deploy and fund ERC20 to Parent and Child chains
       const l1ERC20Token = await deployERC20ToParentChain()
+
+      // Approve custom fee token if not ETH
+      if (isCustomFeeToken) {
+        await approveCustomFeeToken({
+          signer: localWallet.connect(parentProvider),
+          erc20ParentAddress: l1ERC20Token.address
+        })
+        await approveCustomFeeToken({
+          signer: localWallet.connect(parentProvider),
+          erc20ParentAddress: erc20Bridger.nativeToken!
+        })
+        await ethBridger.approveGasToken({
+          parentSigner: localWallet.connect(parentProvider)
+        })
+      }
+      if (isCustomFeeToken) {
+        await fundUserWalletNativeCurrency()
+      }
+
       await fundErc20ToParentChain(l1ERC20Token)
-      await fundErc20ToChildChain(l1ERC20Token)
+      await fundErc20ToChildChain({
+        parentSigner: localWallet.connect(parentProvider),
+        parentErc20Address: l1ERC20Token.address,
+        amount: parseUnits('5', ERC20TokenDecimals)
+      })
       await approveErc20(l1ERC20Token)
 
+      if (
+        !isNonZeroAddress(l1WethAddress) ||
+        !isNonZeroAddress(l2WethAddress)
+      ) {
+        const wethAddresses = await deployWeth(parentProvider)
+        l1WethAddress = wethAddresses.l1WethAddress
+        l2WethAddress = wethAddresses.l2WethAddress
+      }
+
+      if (isCustomFeeToken) {
+        await approveCustomFeeToken({
+          signer: userWallet.connect(parentProvider),
+          erc20ParentAddress: erc20Bridger.nativeToken!
+        })
+        await ethBridger.approveGasToken({
+          parentSigner: userWallet.connect(parentProvider)
+        })
+        await erc20Bridger.approveGasToken({
+          parentSigner: userWallet.connect(parentProvider),
+          erc20ParentAddress: l1WethAddress
+        })
+      }
+
       // Wrap ETH to test WETH transactions and approve it's usage
       await fundWeth('parentChain')
-      await fundWeth('childChain')
       await approveWeth()
+      if (isCustomFeeToken) {
+        await approveCustomFeeToken({
+          signer: userWallet.connect(parentProvider),
+          erc20ParentAddress: l1WethAddress
+        })
+      }
+
+      await fundErc20ToChildChain({
```

Comment:
We can't fund WETH on child chain because we've been wrapping ETH on child chain to get that. Because we're dealing with custom gas tokens we need to bridge it from the parent chain instead